java/tsfile/src/test/java/org/apache/tsfile/encrypt/AES128TsFileReadWriteTest.java

```
@@ -0,0 +1,273 @@
+        Arrays.asList(
+            TSEncoding.PLAIN,
+            TSEncoding.RLE,
+            TSEncoding.TS_2DIFF,
+            TSEncoding.GORILLA_V1,
+            TSEncoding.GORILLA);
+    for (TSEncoding encoding : encodings) {
+      doubleTest(encoding);
+    }
+  }
+
+  public void doubleTest(TSEncoding encoding) throws IOException, WriteProcessException {
+    writeDataByTSRecord(
+        TSDataType.DOUBLE, (i) -> new DoubleDataPoint("sensor_1", (double) i), encoding);
+    readData((i, field, delta) -> assertEquals(i, field.getDoubleV(), delta));
+  }
+
+  // If no dataPoint in "device_1.sensor_2", it will throws a nomeasurement
+  // exception,
+  // cause no schema in tsfilemetadata anymore.
+  @Test
+  public void readEmptyMeasurementTest() throws IOException, WriteProcessException {
+    try (TsFileWriter tsFileWriter = new TsFileWriter(f, new Schema(), conf)) {
+      // add measurements into file schema
+      tsFileWriter.registerTimeseries(
+          new Path(deviceID), new MeasurementSchema("sensor_1", TSDataType.FLOAT, TSEncoding.RLE));
+      tsFileWriter.registerTimeseries(
+          new Path(deviceID),
+          new MeasurementSchema("sensor_2", TSDataType.INT32, TSEncoding.TS_2DIFF));
+      // construct TSRecord
+      TSRecord tsRecord = new TSRecord(1, deviceID);
+      DataPoint dPoint1 = new FloatDataPoint("sensor_1", 1.2f);
+      tsRecord.addTuple(dPoint1);
+      // write a TSRecord to TsFile
+      tsFileWriter.write(tsRecord);
+    }
+
+    // read example : no filter
+    TsFileSequenceReader reader = new TsFileSequenceReader(path);
+    TsFileReader readTsFile = new TsFileReader(reader);
+    ArrayList<Path> paths = new ArrayList<>();
+    paths.add(new Path(deviceID, "sensor_2", true));
+    QueryExpression queryExpression = QueryExpression.create(paths, null);
+    try {
+      QueryDataSet queryDataSet = readTsFile.query(queryExpression);
+    } catch (IOException e) {
+      // Assert.fail();
```

Why ignore the exception?

---

It's the same with TsfileReadWriteTest